examples/exp_configs/rl/multiagent/multiagent_i210.py

```diff
@@ -49,84 +62,180 @@
-    acceleration_controller=(IDMController, {"a": .3, "b": 2.0, "noise": 0.5}),
-    car_following_params=SumoCarFollowingParams(speed_mode="no_collide"),
-)
-vehicles.add(
-    "av",
-    acceleration_controller=(RLController, {}),
-    num_vehicles=0,
-    color='red'
-)
 
 inflow = InFlows()
-# main highway
-pen_rate = PENETRATION_RATE / 100
-assert pen_rate < 1.0, "your penetration rate is over 100%"
-assert pen_rate > 0.0, "your penetration rate should be above zero"
-inflow.add(
-    veh_type="human",
-    edge="119257914",
-    vehs_per_hour=int(VEH_PER_HOUR_BASE_119257914 * (1 - pen_rate)),
-    # probability=1.0,
-    depart_lane="random",
-    departSpeed=20)
-# # on ramp
-# inflow.add(
-#     veh_type="human",
-#     edge="27414345",
-#     vehs_per_hour=321 * pen_rate,
-#     depart_lane="random",
-#     depart_speed=20)
-# inflow.add(
-#     veh_type="human",
-#     edge="27414342#0",
-#     vehs_per_hour=421 * pen_rate,
-#     depart_lane="random",
-#     depart_speed=20)
-
-# Now add the AVs
-# main highway
-inflow.add(
-    veh_type="av",
-    edge="119257914",
-    vehs_per_hour=int(VEH_PER_HOUR_BASE_119257914 * pen_rate),
-    # probability=1.0,
-    depart_lane="random",
-    depart_speed=20)
-# # on ramp
-# inflow.add(
-#     veh_type="av",
-#     edge="27414345",
-#     vehs_per_hour=int(VEH_PER_HOUR_BASE_27414345 * pen_rate),
-#     depart_lane="random",
-#     depart_speed=20)
-# inflow.add(
-#     veh_type="av",
-#     edge="27414342#0",
-#     vehs_per_hour=int(VEH_PER_HOUR_BASE_27414342 * pen_rate),
-#     depart_lane="random",
-#     depart_speed=20)
-
-NET_TEMPLATE = os.path.join(
-    config.PROJECT_PATH,
-    "examples/exp_configs/templates/sumo/test2.net.xml")
-
-warmup_steps = 0
-if additional_env_params['reroute_on_exit']:
-    warmup_steps = 400
+
+if ON_RAMP:
+    vehicles.add(
+        "human",
+        num_vehicles=0,
+        color="white",
+        lane_change_params=SumoLaneChangeParams(
+            lane_change_mode="strategic",
+        ),
+        acceleration_controller=accel_data,
+        routing_controller=(I210Router, {})
+    )
+    if PENETRATION_RATE > 0.0:
+        vehicles.add(
+            "av",
+            num_vehicles=0,
+            color="red",
+            acceleration_controller=(RLController, {}),
+            routing_controller=(I210Router, {})
+        )
+
+    # inflow.add(
+    #     veh_type="human",
+    #     edge=highway_start_edge,
+    #     vehs_per_hour=inflow_rate,
+    #     departLane="best",
+    #     departSpeed=inflow_speed)
+
+    lane_list = ['0', '1', '2', '3', '4']
+
+    for lane in lane_list:
+        inflow.add(
+            veh_type="human",
+            edge=highway_start_edge,
+            vehs_per_hour=int(inflow_rate * (1 - PENETRATION_RATE)),
+            departLane=lane,
+            departSpeed=inflow_speed)
+
+    inflow.add(
+        veh_type="human",
+        edge="27414345",
```

Comment:
When I set `WANT_GHOST_CELL = True` and `ON_RAMP = True`, I get a KeyError. KeyError: '27414345'. I guess the edges are off? Script was `python train.py multiagent_i210 test`

Reply:
Also happens for `WANT_GHOST_CELL = False` and `ON_RAMP = True`

Reply:
False False is fine. Guess it's an issue with the edges in ON_RAMP

Reply:
resolved